src/mds/MDSRank.cc

```diff
@@ -2958,16 +2958,22 @@ void MDSRank::command_scrub_start(Formatter *f,
   bool force = false;
   bool recursive = false;
   bool repair = false;
+  bool eval_stray_at_root = false;
   for (auto &op : scrubop_vec) {
     if (op == "force")
       force = true;
     else if (op == "recursive")
       recursive = true;
     else if (op == "repair")
       repair = true;
+    else if (op == "eval_stray_at_root" && path == "/")
+      eval_stray_at_root = true;
   }
 
   std::lock_guard l(mds_lock);
+  if (eval_stray_at_root) {
+    mdcache->enqueue_scrub("~mdsdir", tag, force, true, repair, f, on_finish);
+  }
```

Comment:
I don't really see the point of this flag. The operator can just queue a scrub for `path = "~mdsdir"`.

Reply:
You are right, that can be done but understanding your previous comments https://github.com/ceph/ceph/pull/47649#issuecomment-1235401150, https://github.com/ceph/ceph/pull/47649#issuecomment-1235862530 and after discussing with @vshankar  in a 1x1, I thought of providing this flag. Rationale being: provide way to exclude `~mdsdir` (both as convenience and safety concern). Someone might want to scrub everything(scrub + eval stray) when they mention root as path and as the scrub by default doesn't do it(i.e. include ~mdsdir) I think this flag should be a good addition.

Reply:
Yeh. I recommended to add that functionality.

Reply:
+1